### PR TITLE
Change large scale workflow to subclass `DockingWorkflowBase`

### LINKED ISSUE #613: Large scale docking WF inputs should subclass WorkflowInputsBase.
@apayne97 large scale docking WF inputs should also subclass WorkflowInputsBase, ill make a seperate PR.

_Originally posted by @hmacdope in https://github.com/choderalab/asapdiscovery/issues/594#issuecomment-1804971246_

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #613

### CODE REVIEW

File: asapdiscovery-docking/asapdiscovery/docking/workflows/workflows.py [hunk 1]
```diff
@@ -31,6 +31,13 @@ class WorkflowInputsBase(BaseModel):
         False, description="Whether to use the Postera database as the query set."
     )
 
+    postera_upload: bool = Field(
+        False, description="Whether to upload the results to Postera."
+    )
+    postera_molset_name: Optional[str] = Field(
+        None, description="The name of the molecule set to upload to."
+    )
+
```
Comment (apayne97) [inline asapdiscovery-docking/asapdiscovery/docking/workflows/workflows.py:40]: You could put these in their own inputs class (`PosteraUploadInputs` and also subclass it in large and small scale) but I'd rather not have these in the DockingWorkflowInputs bc they are specific to the large and small scale docking workflow?

**Review by apayne97 — CHANGES_REQUESTED** [review]
Yay look at all those saved lines of code!

File: asapdiscovery-docking/asapdiscovery/docking/workflows/workflows.py (hunk 1 above)
Comment (hmacdope) [reply asapdiscovery-docking/asapdiscovery/docking/workflows/workflows.py:40]: Ok sounds good, subclassing seems like a good option.

**Comment by hmacdope** [comment]
@apayne97 I am going to be a bit naughty here and merge on green as I need this for #595. If you have any other feedback, feel free to make an issue and i'll sort it out.

**Review by apayne97 — APPROVED** [review]
LGTM :)

**Pull request merged**